### PR TITLE
Sequence caching and misc updates to SERIAL and CREATE/ALTER SEQUENCE pages

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes https://github.com/cockroachdb/docs/issues/7741.
Fixes https://github.com/cockroachdb/docs/issues/10094.
Fixes https://github.com/cockroachdb/docs/issues/10093.
Coupled with https://github.com/cockroachdb/cockroach/pull/62877.

This PR includes:
- New SQL diagrams for `CREATE SEQUENCE` and `ALTER SEQUENCE`.
- Updated examples and output for `CREATE SEQUENCE` and `ALTER SEQUENCE` pages.
- Docs for sequence caching statement parameters and serial sequence caching session variables/cluster settings.

I intentionally did not include an example for sequence caching, as I am not sure how/that we expose cached values, through system catalogs or otherwise...

To this point, `CACHE` does not appear in the `SHOW CREATE` output:

```
> CREATE SEQUENCE cache_test CACHE 10;
> SHOW CREATE SEQUENCE cache_test;

  table_name |                                       create_statement
-------------+------------------------------------------------------------------------------------------------
  cache_test | CREATE SEQUENCE public.cache_test MINVALUE 1 MAXVALUE 9223372036854775807 INCREMENT 1 START 1
(1 row)
```

### CODE REVIEW

**Comment by ericharmeling** [comment]
@ajwerner 
Adding you for review here, as you were the primary reviewer on https://github.com/cockroachdb/cockroach/pull/56954.

**Comment by ajwerner** [comment]
Big yikes on not showing the cache value. Nice job finding that.

**Comment by ericharmeling** [comment]
@ajwerner Ping on review

**Review by ajwerner — APPROVED** [review]
:lgtm: I'll backport the below PR into 21.1.1 

https://github.com/cockroachdb/cockroach/pull/63548 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10219#-:-MYASBtjDJN2wEqd8FMW:bgtcvac)* status: :shipit: complete! 1 of 0 LGTMs obtained

File: v20.1/serial.md [hunk 1]
```diff
@@ -223,17 +226,16 @@ To experience this for yourself, run through the following example in PostgreSQL
...
+    Since each insert increased the sequence in column `a` by one, the first committed insert got the value `645995170080849921`, and the second committed insert got the value `645995193855180801`. As you can see, the values aren't strictly sequential, and the last value doesn't give an accurate count of rows in the table.
 
-In summary, the `SERIAL` type in PostgreSQL and CockroachDB, and the `AUTO_INCREMENT` type in MySQL, all behave the same in that they do not create strict sequences. CockroachDB will likely create more gaps than these other databases, but will generate these values much faster. An alternative feature, introduced in v2.0, is the [`SEQUENCE`](create-sequence.html).
+In summary, the `SERIAL` type in PostgreSQL and CockroachDB, and the `AUTO_INCREMENT` type in MySQL, all behave the same in that they do not create strict sequences. CockroachDB will likely create more gaps than these other databases, but will generate these values much faster. An alternative feature is the [`SEQUENCE`](create-sequence.html).
```
Comment (ianjevans) [inline v20.1/serial.md:238]:
```suggestion
In summary, the `SERIAL` type in PostgreSQL and CockroachDB, and the `AUTO_INCREMENT` type in MySQL, all behave the same in that they do not create strict sequences. CockroachDB will likely create more gaps than these other databases, but will generate these values much faster. An alternative is to use [`SEQUENCE`](create-sequence.html).
```

File: v20.2/create-sequence.md [hunk 2]
```diff
@@ -129,25 +129,37 @@ In this example, we create a sequence with default settings.
 (1 row)
 ~~~
 
-### Create a sequence with user-defined settings
+### Use a sequence when creating a table
 
-In this example, we create a sequence that starts at -1 and descends in increments of 2.
+In this example, we [create a table](create-table.html), using the `nextval()` function for a [default value](default-value.html), with the `customer_seq` sequence as its input:
```
Comment (ianjevans) [inline v20.2/create-sequence.md:134]:
```suggestion
In this example, we [create a table](create-table.html), using the [`nextval()` function](functions-and-operators.html#sequence-functions) for a [default value](default-value.html), with the `customer_seq` sequence as its input:
```

File: v20.2/create-sequence.md [hunk 3]
```diff
@@ -177,7 +189,73 @@ If a value has been obtained from the sequence in the current session, you can a
...
   currval
 -----------
-        3
+        2
+(1 row)
+~~~
+
+### Set the next value of a sequence
+
+In this example, we're going to change the next value of `customer_seq` using the `setval()` function. Currently, the next value will be `3` (i.e., `2` + `INCREMENT 1`). We will change the next value to `5`.
```
Comment (ianjevans) [inline v20.2/create-sequence.md:198]:
```suggestion
In this example, we're going to change the next value of `customer_seq` using the [`setval()` function](functions-and-operators.html#sequence-functions). Currently, the next value will be `3` (i.e., `2` + `INCREMENT 1`). We will change the next value to `5`.
```

File: v20.2/serial.md [hunk 4]
```diff
@@ -207,17 +216,16 @@ To experience this for yourself, run through the following example in PostgreSQL
...
+    Since each insert increased the sequence in column `a` by one, the first committed insert got the value `645994218934534145`, and the second committed insert got the value `645994218978082817`. As you can see, the values aren't strictly sequential, and the last value doesn't give an accurate count of rows in the table.
 
-In summary, the `SERIAL` type in PostgreSQL and CockroachDB, and the `AUTO_INCREMENT` type in MySQL, all behave the same in that they do not create strict sequences. CockroachDB will likely create more gaps than these other databases, but will generate these values much faster. An alternative feature, introduced in v2.0, is the [`SEQUENCE`](create-sequence.html).
+In summary, the `SERIAL` type in PostgreSQL and CockroachDB, and the `AUTO_INCREMENT` type in MySQL, all behave the same in that they do not create strict sequences. CockroachDB will likely create more gaps than these other databases, but will generate these values much faster. An alternative feature is the [`SEQUENCE`](create-sequence.html).
```
Comment (ianjevans) [inline v20.2/serial.md:228]:
```suggestion
In summary, the `SERIAL` type in PostgreSQL and CockroachDB, and the `AUTO_INCREMENT` type in MySQL, all behave the same in that they do not create strict sequences. CockroachDB will likely create more gaps than these other databases, but will generate these values much faster. An alternative is to use [`SEQUENCE`](create-sequence.html).
```

File: v21.1/create-sequence.md [hunk 5]
```diff
@@ -128,25 +129,37 @@ In this example, we create a sequence with default settings.
 (1 row)
 ~~~
 
-### Create a sequence with user-defined settings
+### Use a sequence when creating a table
 
-In this example, we create a sequence that starts at -1 and descends in increments of 2.
+In this example, we [create a table](create-table.html), using the `nextval()` function for a [default value](default-value.html), with the `customer_seq` sequence as its input:
```
Comment (ianjevans) [inline v21.1/create-sequence.md:134]:
```suggestion
In this example, we [create a table](create-table.html), using the [`nextval()` function](functions-and-operators.html#sequence-functions) for a [default value](default-value.html), with the `customer_seq` sequence as its input:
```

File: v21.1/create-sequence.md [hunk 6]
```diff
@@ -176,7 +189,73 @@ If a value has been obtained from the sequence in the current session, you can a
...
   currval
 -----------
-        3
+        2
+(1 row)
+~~~
+
+### Set the next value of a sequence
+
+In this example, we're going to change the next value of `customer_seq` using the `setval()` function. Currently, the next value will be `3` (i.e., `2` + `INCREMENT 1`). We will change the next value to `5`.
```
Comment (ianjevans) [inline v21.1/create-sequence.md:198]:
```suggestion
In this example, we're going to change the next value of `customer_seq` using the [`setval()` function](functions-and-operators.html#sequence-functions). Currently, the next value will be `3` (i.e., `2` + `INCREMENT 1`). We will change the next value to `5`.
```

**Review by ianjevans — APPROVED** [review]
Looks good, just some small nits and added links.

**Review by ericharmeling — COMMENTED** [review]
TFTR @ianjevans !

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10219#-:-MYomv8J08bfXgoLtq6S:b-vo6kmr)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ianjevans)

___
*[v20.1/serial.md, line 238 at r1](https://reviewable.io/reviews/cockroachdb/docs/10219#-MX8x5cDI_bP5ZGAb1Gu-r1-238:-MYon3OmE3T-A6sK7Msc:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/761c6b77b522b7f5aaac17a926fce47184611714/v20.1/serial.md#L238)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
In summary, the `SERIAL` type in PostgreSQL and CockroachDB, and the `AUTO_INCREMENT` type in MySQL, all behave the same in that they do not create strict sequences. CockroachDB will likely create more gaps than these other databases, but will generate these values much faster. An alternative is to use [`SEQUENCE`](create-sequence.html).
```
</blockquote></details>

Done.
___
*[v20.2/create-sequence.md, line 134 at r1](https://reviewable.io/reviews/cockroachdb/docs/10219#-MX8x5cDI_bP5ZGAb1Gw-r1-134:-MYonb2Q16uEhdL2pX8M:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/761c6b77b522b7f5aaac17a926fce47184611714/v20.2/create-sequence.md#L134)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
In this example, we [create a table](create-table.html), using the [`nextval()` function](functions-and-operators.html#sequence-functions) for a [default value](default-value.html), with the `customer_seq` sequence as its input:
```
</blockquote></details>

Done.
___
*[v20.2/create-sequence.md, line 198 at r1](https://reviewable.io/reviews/cockroachdb/docs/10219#-MX8x5cDI_bP5ZGAb1Gw-r1-198:-MYont8j-ofLxfYhkjUT:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/761c6b77b522b7f5aaac17a926fce47184611714/v20.2/create-sequence.md#L198)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
In this example, we're going to change the next value of `customer_seq` using the [`setval()` function](functions-and-operators.html#sequence-functions). Currently, the next value will be `3` (i.e., `2` + `INCREMENT 1`). We will change the next value to `5`.
```
</blockquote></details>

Done.
___
*[v20.2/serial.md, line 228 at r1](https://reviewable.io/reviews/cockroachdb/docs/10219#-MX8x5cDI_bP5ZGAb1Gx-r1-228:-MYonu2H4PiYApi70vtI:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/761c6b77b522b7f5aaac17a926fce47184611714/v20.2/serial.md#L228)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
In summary, the `SERIAL` type in PostgreSQL and CockroachDB, and the `AUTO_INCREMENT` type in MySQL, all behave the same in that they do not create strict sequences. CockroachDB will likely create more gaps than these other databases, but will generate these values much faster. An alternative is to use [`SEQUENCE`](create-sequence.html).
```
</blockquote></details>

Done.
___
*[v21.1/create-sequence.md, line 134 at r1](https://reviewable.io/reviews/cockroachdb/docs/10219#-MX8x5cDI_bP5ZGAb1Gz-r1-134:-MYoo4O56uSz-MRNCet_:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/761c6b77b522b7f5aaac17a926fce47184611714/v21.1/create-sequence.md#L134)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
In this example, we [create a table](create-table.html), using the [`nextval()` function](functions-and-operators.html#sequence-functions) for a [default value](default-value.html), with the `customer_seq` sequence as its input:
```
</blockquote></details>

Done.
___
*[v21.1/create-sequence.md, line 198 at r1](https://reviewable.io/reviews/cockroachdb/docs/10219#-MX8x5cDI_bP5ZGAb1Gz-r1-198:-MYoo3XZ8OBMzZ_U3vss:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/761c6b77b522b7f5aaac17a926fce47184611714/v21.1/create-sequence.md#L198)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
In this example, we're going to change the next value of `customer_seq` using the [`setval()` function](functions-and-operators.html#sequence-functions). Currently, the next value will be `3` (i.e., `2` + `INCREMENT 1`). We will change the next value to `5`.
```
</blockquote></details>

Done.

**Pull request merged**